src/Plugins/ComplexCore/src/ComplexCore/Filters/PartitionGeometryFilter.cpp

```diff
@@ -0,0 +1,533 @@
+                                                                    ChoicesParameter::Choices{"Basic", "Advanced", "Bounding Box", "Existing Partitioning Scheme"}));
+  params.insert(std::make_unique<Int32Parameter>(k_StartingPartitionID_Key, "Starting Partition ID", "", 1));
+  params.insert(std::make_unique<Int32Parameter>(k_OutOfBoundsValue_Key, "Out-Of-Bounds Partition ID", "", 0));
+  params.insert(std::make_unique<VectorInt32Parameter>(k_NumberOfPartitionsPerAxis_Key, "Number Of Partitions Per Axis (X, Y, Z)", "", std::vector<int32>(3), std::vector<std::string>(3)));
+  params.insert(std::make_unique<VectorFloat32Parameter>(k_PartitioningSchemeOrigin_Key, "Partitioning Scheme Origin (X, Y, Z)", "", std::vector<float32>(3), std::vector<std::string>(3)));
+  params.insert(std::make_unique<VectorFloat32Parameter>(k_LengthPerPartition_Key, "Length Per Partition (X, Y, Z)", "", std::vector<float32>(3), std::vector<std::string>(3)));
+  params.insert(std::make_unique<VectorFloat32Parameter>(k_LowerLeftCoord_Key, "Lower Left Coordinate (X, Y, Z)", "", std::vector<float32>(3), std::vector<std::string>(3)));
+  params.insert(std::make_unique<VectorFloat32Parameter>(k_UpperRightCoord_Key, "Upper Right Coordinate (X, Y, Z)", "", std::vector<float32>(3), std::vector<std::string>(3)));
+  params.insert(std::make_unique<GeometrySelectionParameter>(k_ExistingPartitioningSchemePath_Key, "Existing Partitioning Scheme", "", DataPath{},
+                                                             GeometrySelectionParameter::AllowedTypes{IGeometry::Type::Image}));
+
+  params.insertLinkableParameter(std::make_unique<BoolParameter>(k_UseVertexMask_Key, "Use Vertex Mask", "", false));
+  params.insert(std::make_unique<ArraySelectionParameter>(k_VertexMaskPath_Key, "Vertex Mask", "", DataPath{}, ArraySelectionParameter::AllowedTypes{DataType::boolean},
+                                                          ArraySelectionParameter::AllowedComponentShapes{{1}}));
+
+  params.insertSeparator(Parameters::Separator{"Created Geometry Objects"});
+  params.insert(std::make_unique<StringParameter>(k_FeatureAttrMatrixName_Key, "Feature Attribute Matrix Name", "", "Feature Data"));
+  params.insert(std::make_unique<StringParameter>(k_PartitionIdsArrayName_Key, "Partition Ids Array Name", "", "Partition Ids"));
+
+  params.insertSeparator(Parameters::Separator{"Created Partitioning Scheme Objects"});
+  params.insert(std::make_unique<DataGroupCreationParameter>(k_PSGeometry_Key, "Partitioning Scheme Geometry", "The complete path to the Partitioning Scheme Geometry being created",
+                                                             DataPath({"[Partitioning Scheme Geometry]"})));
+  params.insert(std::make_unique<StringParameter>(k_PSGeometryAMName_Key, "Partitioning Scheme Attribute Matrix", "", "Cell Data"));
+  params.insert(std::make_unique<StringParameter>(k_PSGeometryDataName_Key, "Partitioning Scheme Ids", "", "PartitioningSchemeIds"));
```

Comment:
If these are creating DataObjects that get inserted into the DataStructure I think we should use the DataObjectNameParameter instead of StringParameter